modules/platforms/dotnet/Apache.Ignite.Core/Datastream/IDataStreamer.cs

```diff
@@ -168,55 +184,90 @@ public interface IDataStreamer<TK, TV> : IDisposable
+        /// Adds collection of key-value pairs for loading.
         /// </summary>
         /// <param name="entries">Entries.</param>
         /// <returns>Task for this operation.</returns>
+        [Obsolete("Use Add.")]
         Task AddData(ICollection<KeyValuePair<TK, TV>> entries);
 
         /// <summary>
         /// Adds key for removal.
         /// </summary>
         /// <param name="key">Key.</param>
         /// <returns>Task for this operation.</returns>
+        [Obsolete("Use Remove.")]
         Task RemoveData(TK key);
 
         /// <summary>
-        /// Makes an attempt to load remaining data. This method is mostly similar to 
-        /// <see cref="IDataStreamer{K,V}.Flush()"/> with the difference that it won't wait and 
+        /// Adds single key-value pair for loading. Passing <c>null</c> as value will be
+        /// interpreted as removal.
+        /// </summary>
+        /// <param name="key">Key.</param>
+        /// <param name="val">Value.</param>
+        void Add(TK key, TV val);
+
+        /// <summary>
+        /// Adds single key-value pair for loading. Passing <c>null</c> as pair's value will
+        /// be interpreted as removal.
+        /// </summary>
+        /// <param name="pair">Key-value pair.</param>
+        void Add(KeyValuePair<TK, TV> pair);
+
+        /// <summary>
+        /// Adds collection of key-value pairs for loading.
+        /// </summary>
+        /// <param name="entries">Entries.</param>
+        void Add(ICollection<KeyValuePair<TK, TV>> entries);
+
+        /// <summary>
+        /// Adds key for removal.
+        /// </summary>
+        /// <param name="key">Key.</param>
+        void Remove(TK key);
+
+        /// <summary>
+        /// Makes an attempt to load remaining data. This method is mostly similar to
+        /// <see cref="IDataStreamer{K,V}.Flush()"/> with the difference that it won't wait and
         /// will exit immediately.
         /// </summary>
+        [Obsolete("Use FlushAsync")]
         void TryFlush();
 
         /// <summary>
-        /// Loads any remaining data, but doesn't close the streamer. Data can be still added after
-        /// flush is finished. This method blocks and doesn't allow to add any data until all data
-        /// is loaded.
+        /// Loads any remaining buffered data, but doesn't close the streamer.
         /// </summary>
         void Flush();
 
+        /// <summary>
+        /// Loads any remaining buffered data, but doesn't close the streamer.
+        /// </summary>
+        Task FlushAsync();
+
         /// <summary>
         /// Closes this streamer optionally loading any remaining data.
         /// </summary>
         /// <param name="cancel">Whether to cancel ongoing loading operations. When set to <c>true</c>
-        /// there is not guarantees what data will be actually loaded to cache.</param>
+        /// there are no guarantees what data will be actually loaded to cache.</param>
```

Comment:
Actually, "there is no guarantee" sounds more authentic to me.

Reply:
Fixed